### PR TITLE
Invoices

### LINKED ISSUE #8: Generate Invoices
Generating invoices for a department needs to be built and tested, we may wait on a second or third pass to implement this

### LINKED ISSUE #148: Style Change home page borders
- Spacing of the number (of rentals) and text
- Borders around the table, we want only a top-border.

### LINKED ISSUE #149: Updating the Rental show page
- no spire ID in show
- display reservation number
- instead of uuid use a reservation id

### PR DESCRIPTION
**Pull request by tomecho**
fixes #8 
fixes #148 
fixes #149

### CODE REVIEW

File: app/controllers/rentals_controller.rb [hunk 1]
```diff
@@ -98,13 +94,21 @@ def destroy
...
+
   private
 
   # Use callbacks to share common setup or constraints between actions.
   def set_rental
     @rental = Rental.find(params[:id])
   end
 
+  def set_financial_transactions
+    @financial_transactions = FinancialTransaction.where(rental_id: @rental.id)
```
Comment (phantompatrol) [inline app/controllers/rentals_controller.rb]: Perhaps: `FinancialTransaction.where(rental: @rental)`

File: app/views/rentals/invoice.html.erb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+      </table>
+    </div>
+  </div>
+
+  <div id="payment" class="tab-pane fade">
+    <%= render partial: "transaction_detail" %>
+  </div>
+</div>
+
+<p style="margin-bottom: 15px">This invoice may be paid by Procard on-line or at our office or by recharge form. <br>
```
Comment (phantompatrol) [inline app/views/rentals/invoice.html.erb]: Please move to the style into the asset pipeline

File: spec/controllers/rentals_controller_spec.rb [hunk 3]
```diff
@@ -156,15 +156,27 @@
...
 
+  describe 'GET #invoice' do
+    it 'assigns a requested rental to @rental' do
+      get :invoice, params: { id: @rental }
+      expect(assigns[:rental]).to eq @rental
+    end
+
+    it 'all requested financial transactions should contain the same rental as @rental' do
+      get :show, params: { id: @rental }
+      expect(assigns[:financial_transactions].all? { |ft| ft.rental.id == @rental.id }).to be true
```
Comment (phantompatrol) [inline spec/controllers/rentals_controller_spec.rb]: Does set difference work `expect(assigns[:financial_transactions].map(:rental_id) - [@rental.id]).to be_empty`
work by any chance?
Or perhaps`expect(assigns[:financial_transactions].map(:rental_id).uniq).to eq([@rental.id])`

File: config/initializers/assets.rb [hunk 4]
```diff
@@ -10,4 +10,5 @@
 # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
 Rails.application.config.assets.precompile += %w( rentals/index.css )
 Rails.application.config.assets.precompile += %w( rentals/index.js )
+Rails.application.config.assets.precompile += %w( rentals/invoice.css )
```
Comment (tomecho) [inline config/initializers/assets.rb]: why not `config.assets.precompile = ['*.js', '*.css']`?

**Comment by tomecho** [comment]
fixed

**Comment by tomecho** [comment]
fixing visual glitch on invoices

**Comment by tomecho** [comment]
As far as i can tell chrome tries to reprint the table header at the top of every page if it's printing a continued table. (i tried using page-break-inside: avoid, it confuses chrome more)

**Pull request merged**